cognify/optimizer/evaluator.py

```diff
@@ -477,29 +486,31 @@ def get_score(
         sema = mp.Semaphore(n_parallel)
         result_q = mp.Queue()
 
-        total_score, total_cost, n_success = 0.0, 0.0, 0
+        total_score, total_cost, total_exec_time, n_success = 0.0, 0.0, 0.0, 0
         opt_trace = ".".join(task.trace_back)
 
         def update_pbar(pbar, eval_result):
-            nonlocal total_score, total_cost, n_success
+            nonlocal total_score, total_cost, total_exec_time, n_success
             n_success += 1
-            score, price = eval_result[3], eval_result[4]
+            score, price, exec_time = eval_result[3], eval_result[4], eval_result[6]   # using these indices is dangerous
```

Comment:
don't use hard-coded indices

Reply:
Apply weight to different objectives during the search is non-trivial (also not straightforward for users to set), maybe we can add on/off in this PR.

Reply:
Done (removed hardcoded indices). Instead, it's an actual data structure. As for allowing users to select which one, I am adding that in a separate PR. This PR will just jointly optimize quality, cost, and latency and report that to the user.

Reply:
Btw, see #23 for allowing users to select on/off for different targets (e.g., `["cost", "latency"]` or `["quality", "latency"]`). That one includes the changes made here as well, so if you want to review that one and just merge that, then we can close this. Otherwise, we can merge them 1 at a time.